### PR TITLE
Report the git commit hash in groupversion_info_gen.go

### LINKED ISSUE #118: Improvement: Include generator version in generated files
To allow code generation issues to be correctly traced back to the responsible version of the code generator, we desire for the files to be stamped with the version of the generator.

Marking all files would result in all files showing as changed when regenerated, which isn't good for tracking actual changes. Instead, let's embed the code generator version in the `groupversion_info.go`. file

### PR DESCRIPTION
**Pull request by babbageclunk**
This will help us trace a generated package back to the code that produced it. The git commit hash (and whether the tree is dirty) is fed into the build from the Makefile.

Fixes #118

### CODE REVIEW

**Comment by babbageclunk** [comment]
Ha, got CLA'd.

File: hack/generator/Makefile [hunk 1]
```diff
@@ -19,7 +20,7 @@ manifests: $(CONTROLLER_GEN)
 
 .PHONY: gen-arm
 gen-arm: rm-apis build ; $(info $(M) running k8sinfra-gen gen…) @ ## Running gen
-	$(Q) ./bin/$(APP) gen azure-arm.yaml
+	$(Q) ./bin/$(APP) -v 1 gen azure-arm.yaml
```
Comment (matthchr) [inline hack/generator/Makefile:23]: This is setting log verbosity right?

Is there a reason we want more than the default level of logging during the build?
Phrased another way: if there is information we think is useful to be displayed all the time would we rather just put that stuff in v=0 (which I think is the default?)

File: hack/generator/pkg/codegen/version.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+func combinedVersion() string {
+	result := GitCommit
+	if GitTreeState != "clean" {
+		result += fmt.Sprintf(" (tree is %s)", GitTreeState)
+	}
+	return result
+
+}
```
Comment (Porges) [inline hack/generator/pkg/codegen/version.go:25]: minor: some odd formatting here?

File: hack/generator/Makefile (hunk 1 above)
Comment (babbageclunk) [reply hack/generator/Makefile:23]: Agreed - changing it back.

File: hack/generator/pkg/codegen/version.go (hunk 2 above)
Comment (babbageclunk) [reply hack/generator/pkg/codegen/version.go:25]: Doh, fixed.

**Pull request merged**